classes/Carrier.php

```diff
@@ -1390,7 +1390,9 @@ public static function sqlDeliveryRangeShop($range_table, $alias = 'd')
      */
     public function updatePosition($way, $position)
     {
-        if (!$res = Db::getInstance()->executeS(
+        $db = Db::getInstance();
+
+        if (!$res = $db->executeS(
```

Comment:
You need to use a try/catch block here Whenever we open a transaction the whole process should always be wrapped in a try/catch block so that you can rollback in case an unexpected exception is triggered. Something like:

```php
$db = Db::getInstance();
try {
  $db->beginTransaction();
  // Do any queries
  $db->commit();
} catch (Exception $e) {
  $db->rollback();
  throw $e;
}
```

We never block the exception we just make sure that we always rollback before.